csharp/rust/src/lib.rs

```diff
@@ -1,284 +1,472 @@
-    let runtime = Builder::new_multi_thread()
-        .enable_all()
-        .thread_name("GLIDE C# thread")
-        .build()?;
-    let _runtime_handle = runtime.enter();
-    let client = runtime.block_on(GlideClient::new(request, None)).unwrap(); // TODO - handle errors.
-    let core = Arc::new(CommandExecutionCore {
-        success_callback,
-        failure_callback,
-        client,
-    });
-    Ok(Client { runtime, core })
-}
-
-/// Creates a new client with the given configuration.
-/// The success callback needs to copy the given string synchronously, since it will be dropped by Rust once the callback returns.
-/// All callbacks should be offloaded to separate threads in order not to exhaust the client's thread pool.
+/// # Parameters
+/// - ***in_minimal_level*** The minimum file log level
+/// - ***in_file_name*** The file name to log to
 ///
-/// # Safety
+/// # Input Safety (in_...)
+/// The data passed in is considered "caller responsibility".
+/// Any pointers hence will be left unreleased after leaving.
 ///
-/// * `config` must be a valid [`ConnectionConfig`] pointer. See the safety documentation of [`create_client_internal`].
-#[allow(rustdoc::private_intra_doc_links)]
-#[no_mangle]
-pub unsafe extern "C" fn create_client(
-    config: *const ConnectionConfig,
-    success_callback: unsafe extern "C" fn(usize, i32, *const c_char) -> (),
-    failure_callback: unsafe extern "C" fn(usize) -> (),
-) -> *const c_void {
-    match unsafe { create_client_internal(config, success_callback, failure_callback) } {
-        Err(_) => std::ptr::null(), // TODO - log errors
-        Ok(client) => Arc::into_raw(Arc::new(client)) as *const c_void,
-    }
-}
-
-/// Closes the given client, deallocating it from the heap.
-/// This function should only be called once per pointer created by [`create_client`].
-/// After calling this function the `client_ptr` is not in a valid state.
+/// # Output Safety (out_... / return ...)
+/// The data returned is considered "caller responsibility".
+/// The caller must release any non-null pointers.
 ///
-/// # Safety
+/// # Reference Safety (ref_...)
+/// Any reference data is considered "caller owned".
 ///
-/// * `client_ptr` must not be `null`.
-/// * `client_ptr` must be able to be safely casted to a valid [`Box<Client>`] via [`Box::from_raw`]. See the safety documentation of [`std::boxed::Box::from_raw`].
+/// # Freeing data allocated by the API
+/// To free data returned by the API, use the corresponding `free_...` methods of the API.
+/// It is **not optional** to call them to free data allocated by the API!
 #[no_mangle]
-pub extern "C" fn close_client(client_ptr: *const c_void) {
-    assert!(!client_ptr.is_null());
-    // This will bring the strong count down to 0 once all client requests are done.
-    unsafe { Arc::decrement_strong_count(client_ptr as *const Client) };
-}
+pub extern "C-unwind" fn csharp_system_init(
+    in_minimal_level: ELoggerLevel,
```

Comment:
Logger could be reconfigured on the fly.
Consider to drop/postpone logger-related changes to shorten this PR